rustls-libssl/src/constants.rs

```diff
@@ -83,3 +87,18 @@ pub fn alert_desc_to_short_string(value: c_int) -> &'static CStr {
         _ => c"UK",
     }
 }
+
+pub fn sig_scheme_to_nid(scheme: SignatureScheme) -> Option<c_int> {
+    use SignatureScheme::*;
+    match scheme {
+        RSA_PKCS1_SHA256 | RSA_PKCS1_SHA384 | RSA_PKCS1_SHA512 => Some(NID_rsaEncryption),
+        RSA_PSS_SHA256 | RSA_PSS_SHA384 | RSA_PSS_SHA512 => Some(NID_rsassaPss),
```

Comment:
I read your [other comment](https://github.com/rustls/rustls-openssl-compat/commit/a79305d9ca99db6768507b7d9510e1f3e506d881#r141636079) (and part of the man page) as suggesting the PSS schemes should also return `NID_rsaEncryption`, but in practice doing so was breaking the client/server tests - real OpenSSL was producing `912` (`NID_rsassaPss`).

Reply:
Ah, good stuff. I see that special case is mentioned on `SSL_get_peer_signature_type_nid(3ssl)` so that makes sense.